File: app/src/main/java/timothypaetz/com/recyclersectionheader/Person.java
package timothypaetz.com.recyclersectionheader;

import android.support.annotation.NonNull;
import java.util.Locale;

/**
 * Created by paetztm on 2/6/2017.
 */

public class Person implements Comparable<Person> {
    private final CharSequence firstName;
    private final CharSequence lastName;
    private static final String NAME_DISPLAY = "%s, %s";

    public Person(@NonNull CharSequence firstName, @NonNull CharSequence lastName) {
        this.firstName = firstName;
        this.lastName = lastName;
    }

    public CharSequence getLastName() {
        return lastName;
    }

    public CharSequence getFirstName() {
        return firstName;
    }

    public String getFullName() {
        return String.format(Locale.getDefault(),
                             NAME_DISPLAY,
                             getLastName(),
                             getFirstName());
    }

    @Override
    public int compareTo(@NonNull Person person) {
        return getLastName().toString()
                            .compareTo(person.getLastName()
                                             .toString());
    }
}

File: app/src/main/java/timothypaetz/com/recyclersectionheader/Person.kt
package timothypaetz.com.recyclersectionheader

import java.util.*

/**
 * Created by paetztm on 2/6/2017.
 */
data class Person(val firstName: CharSequence, val lastName: CharSequence) : Comparable<Person> {
    val fullName = String.format(
            Locale.getDefault(),
            NAME_DISPLAY,
            lastName,
            firstName
        )

    override fun compareTo(other: Person): Int {
        return lastName.toString()
            .compareTo(
                other.lastName
                    .toString()
            )
    }

    companion object {
        private const val NAME_DISPLAY = "%s, %s"
    }
}